src/api/SocialTradingAPI.ts

```diff
@@ -660,6 +705,23 @@ export class SocialTradingAPI {
     } as TradingPoolRebalanceInfo;
   }
 
+  private createTradingPoolAccumulationObject(
+    tradingPoolFees: any,
+  ): TradingPoolAccumulationInfo[] {
+    const streamingFees = tradingPoolFees[0];
+    const profitFees = tradingPoolFees[1];
+
+    return streamingFees.reduce((accumulator: [], streamingFee: BigNumber, index: number) => {
```

Comment:
Awkward implementation affecting readability. Do a `for` loop of the length of one of them, iterate and build a hash by accessing the same index of both arrays at the same time.